dpctl/_sycl_platform.pyx

```diff
@@ -366,6 +375,78 @@ cdef class SyclPlatform(_SyclPlatform):
         """
         return DPCTLPlatform_Hash(self._platform_ref)
 
+    def get_devices(self, device_type=device_type_t.all):
+        """
+        Returns the list of :class:`dpctl.SyclDevice` objects associated with
+        :class:`dpctl.SyclPlatform` instance selected based on
+        the given :class:`dpctl.device_type`.
+
+        Args:
+            device_type (optional):
+                A :class:`dpctl.device_type` enum value or a string that
+                specifies a SYCL device type. Currently, accepted values are:
+                "gpu", "cpu", "accelerator", or "all".
+                Default: ``dpctl.device_type.all``.
+
+        Returns:
+            list:
+                A :obj:`list` of :class:`dpctl.SyclDevice` objects
+                that belong to this platform.
+
+        Raises:
+            TypeError:
+                If `device_type` is not a str or :class:`dpctl.device_type`
```

Comment:
```suggestion
                If `device_type` is not a string or :class:`dpctl.device_type`
```

Reply:
It appears the enum types defined in `enum_types.py` are not documented anywhere. They will need a page similar to the constants page for `dpctl.tensor`

Good catch!